Add docs for smart retry (#1939)
saucelabs/sauce-docs: docs/web-apps/automated-testing/cucumberjs-playwright/yaml.md

---

### `smartRetry`

<p><small>| OPTIONAL | OBJECT |</small></p>

> Specifies the retry strategy to apply for that suite. It should be used along with [retries](#retries).

Reviewer: "should" in this context would mean that it's a strong recommendation. But it's not. It won't do anything _unless_ this feature is used with retries.

```suggestion
Specifies the retry strategy to apply for that suite. Requires [retries](#retries) to be >= 1.
```

```yaml
sauce:
  retries: 3
suite:
  - name: My Saucy Test
    smartRetry:
      failedOnly: true
```

---

#### `failedOnly`

<p><small>| OPTIONAL | BOOLEAN |</small></p>

> When set to `true`, the retry will test only the spec files that failed during the previous try.

Reviewer: ```suggestion
When set to `true`, only the spec files that failed during the previous attempt are retried.
```
